lnwallet/channel.go

```diff
@@ -6636,12 +6746,42 @@ func newOutgoingHtlcResolution(signer input.Signer,
 	// Finally, we'll generate the script output that the timeout
 	// transaction creates so we can generate the signDesc required to
 	// complete the claim process after a delay period.
-	htlcSweepScript, err := SecondLevelHtlcScript(
-		chanType, isCommitFromInitiator, keyRing.RevocationKey,
-		keyRing.ToLocalKey, csvDelay, leaseExpiry,
+	var (
+		htlcSweepScript *ScriptInfo
+		signMethod      input.SignMethod
+		ctrlBlock       []byte
 	)
-	if err != nil {
-		return nil, err
+	if !chanType.IsTaproot() {
```

Comment:
hmm why the change of pattern `if chanType.IsTaproot()`?

Reply:
Both are used in the diff. Only other location of the positive check is when we go to parse the ecdsa vs schnorr signature in `newIncomingHtlcResolution` (outgoing one too).